### ISSUE TITLE
Developer warnings

### ISSUE THREAD

**Issue by eike-fokken** [scivision/mumps-superbuild#19]
I have no idea what they mean, but running `cmake -S . -B build` inside the root directory of the repo yields these warnings (although everything seems to work otherwise)

```sh
-- Found LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so;/usr/lib/x86_64-linux-gnu/blas/libblas.so;-lpthread  found components: Netlib 
CMake Warning (dev) at /usr/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (PkgConfig)
  does not match the name of the calling package (MPI).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  /usr/share/cmake-3.21/Modules/FindPkgConfig.cmake:70 (find_package_handle_standard_args)
  /usr/share/cmake-3.21/Modules/FindMPI.cmake:270 (include)
  CMakeLists.txt:58 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found MPI_C: /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so (found version "3.1") 
-- Found MPI_Fortran: /usr/lib/x86_64-linux-gnu/libmpi_usempif08.so (found version "3.1") 
-- Found MPI: TRUE (found version "3.1") found components: C Fortran 
-- Checking for one of the modules 'scalapack;scalapack-openmpi;scalapack-mpich'
CMake Warning (dev) at /usr/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (PkgConfig)
  does not match the name of the calling package (MPI).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  /usr/share/cmake-3.21/Modules/FindPkgConfig.cmake:70 (find_package_handle_standard_args)
  /usr/share/cmake-3.21/Modules/FindMPI.cmake:270 (include)
  cmake/Modules/FindSCALAPACK.cmake:64 (find_package)
  cmake/Modules/FindSCALAPACK.cmake:206 (scalapack_check)
  cmake/scalapack.cmake:20 (find_package)
  CMakeLists.txt:59 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

Just wanted to let you know. I'm on Linux and my cmake is `cmake version 3.21.3`.

**Comment by scivision** [comment]
I think this will be fixed in CMake 3.22.0 release, coming in ~ several days: https://gitlab.kitware.com/cmake/cmake/-/issues/22823

**Comment by eike-fokken** [comment]
Ah, nice!
